cpp/src/traversal/mg/common_utils.cuh

```diff
@@ -102,20 +115,401 @@ struct bfs_frontier_pred_dist {
+  if (block_acceptable_vertex_count == 0) { return; }
+
+  if (threadIdx.x == 0) {
+    block_offset = cugraph::detail::traversal::atomicAdd(
+      out_count, static_cast<T>(block_acceptable_vertex_count));
+  }
+  __syncthreads();
+
+  if (acceptable_vertex) { out_id[block_offset + thread_write_offset] = id; }
+}
+
+template <unsigned BLOCK_SIZE, typename T>
+__global__ void remove_duplicates_kernel(unsigned *bmap,
+                                         unsigned *isolated_bmap,
+                                         T *in_id,
+                                         T id_begin,
+                                         T id_end,
+                                         T count,
+                                         T *out_id,
+                                         T *out_count)
+{
+  T tid = blockIdx.x * blockDim.x + threadIdx.x;
+  T id;
+  if (tid < count) {
+    id = in_id[tid];
+  } else {
+    // Invalid vertex id to avoid partial thread block execution
+    id = id_end;
+  }
+
+  int acceptable_vertex = 0;
+  // If id is not in the acceptable range then set it to
+  // an invalid vertex id
+  if ((id >= id_begin) && (id < id_end)) {
+    unsigned active_bit = static_cast<unsigned>(1) << (id % BitsPWrd<unsigned>);
+    unsigned prev_word  = atomicOr(bmap + (id / BitsPWrd<unsigned>), active_bit);
+    // If bit was set by this thread then the id is unique
+    if (!(prev_word & active_bit)) {
+      // If id is isolated (out-degree == 0) then mark it as unacceptable
+      bool is_dst_isolated = active_bit & isolated_bmap[id / BitsPWrd<unsigned>];
+      acceptable_vertex    = !is_dst_isolated;
+    }
+  }
+
+  __shared__ T block_offset;
+  typedef cub::BlockScan<int, BLOCK_SIZE> BlockScan;
+  __shared__ typename BlockScan::TempStorage temp_storage;
+  int thread_write_offset;
+  int block_acceptable_vertex_count;
+  BlockScan(temp_storage)
+    .ExclusiveSum(acceptable_vertex, thread_write_offset, block_acceptable_vertex_count);
+
+  // If the block is not going to write unique ids then return
+  if (block_acceptable_vertex_count == 0) { return; }
+
+  if (threadIdx.x == 0) {
+    block_offset = cugraph::detail::traversal::atomicAdd(
+      out_count, static_cast<T>(block_acceptable_vertex_count));
+  }
+  __syncthreads();
+
+  if (acceptable_vertex) { out_id[block_offset + thread_write_offset] = id; }
+}
+
+template <typename T>
+T remove_duplicates(raft::handle_t const &handle,
+                    rmm::device_vector<unsigned> &bmap,
+                    rmm::device_vector<T> &data,
+                    T data_len,
+                    T data_begin,
+                    T data_end,
+                    rmm::device_vector<T> &out_data)
+{
+  cudaStream_t stream = handle.get_stream();
+
+  rmm::device_vector<T> unique_count(1, 0);
+
+  thrust::fill(
+    rmm::exec_policy(stream)->on(stream), bmap.begin(), bmap.end(), static_cast<unsigned>(0));
+  T threads = 256;
```

Comment:
This should be `constexpr T threads{256};`

Reply:
Interesting find. The following fails to compile. I get a parsing error in the line _before_ the `constexpr` line.
```
  constexpr return_t threads{256};
  return_t blocks  = raft::div_rounding_up_safe(data_len, threads);
  remove_duplicates_kernel<threads><<<blocks, threads, 0, stream>>>(bmap.data().get(), ...
```
This one though compiles fine.

```
  constexpr return_t threads = 256;
  return_t blocks  = raft::div_rounding_up_safe(data_len, threads);
  remove_duplicates_kernel<threads><<<blocks, threads, 0, stream>>>(bmap.data().get(), ...

```

Is this expected?

Reply:
Nope. Cannot repro with `g++` on a simple example. Might be `nvcc`? We can follow-up in a separate (Slack) thread, and see if maybe we should raise the issue with the compiler team. But for our purposes, 
`constexpr return_t threads = 256;` should be fine.

Reply:
Well, if you instantiate your call with `return_t == char` you'll get a narrowing conversion error with the brace-initializer. Which would only be a warning for the assignment version. But you should get that on that (`constexpr`) line, not before.

Reply:
Short repro:
```
//nvcc -c -std=c++14 --expt-extended-lambda brace-init-woes.cu
//nvcc -std=c++14 --expt-extended-lambda brace-init-woes.cu -o biw.exe

#include <iostream>
#include <vector>

#include <thrust/device_vector.h>
#include <thrust/reduce.h>
#include <thrust/functional.h>
#include <thrust/transform.h>


using size_type = int;

template <typename return_t,
          typename value_type,
          typename ThrustExecPolicy>
return_t fill(ThrustExecPolicy t_exe_pol, value_type* buffer_,
          size_type size_, value_type value) {
    thrust::fill_n(t_exe_pol, buffer_, size_, value);
    //constexpr return_t threads = 256;//warn
    //constexpr return_t threads(256); //warn
    constexpr return_t threads{256}; //error
    return threads;
}


int main(void)
{
  std::vector<double> v{3.14, -2.13, -1.01, 10.17};
  thrust::device_vector<double> dv(v);

  size_type sz = dv.size();
  auto t = fill<char>(thrust::device, dv.data().get(), sz, 17.13);

  std::cout<<"Done!\n";
  return 0;
}
```